### DISCUSSION TITLE
Problem with Netlify?

### DISCUSSION THREAD

**Discussion by akiarostami** [sveltejs/kit#3782] [Q&A]
I have a very simple app that I used to run on Netlify. At some point and after one of the updates (I don't know which one!), I could not log in to the site anymore. When I call a local endpoint, I get error 500 (server error).

Everything works fine on my local machine, and when I deploy on Netlify, I can't call the endpoint. The strange thing is I can call the  endpoint from postman or any other tool, but calling it from sveltekit on the same site returns error 500.

After days of trying to fix the issue, I decided to try it on Vercel, and everything worked fine! 

You can see the same code running on [Netlify](https://admin.docunight.com) vs [Vercel](https://docunight-admin.vercel.app) (you can enter any email there). Check the browser console for more info. And this is my Netlify toml file (I tried different variations of toml file as well):

```
[build]
  command = "yarn build"
  publish = "build"
  functions = "functions"

[functions]
  directory = "netlify/functions"
  node_bundler = "esbuild"
```

Any idea what is wrong? I hope this is not a coup against Netlify, now that Vercel is backing Svelte... 😁 (just kidding).

**Comment by thesyedbasim** [comment]
I don't know much about it, but have you tried [adapter-netlify](https://github.com/sveltejs/kit/tree/master/packages/adapter-netlify)?

Maybe it can fix your problem.

**Comment by benmccann** [comment]
There's not enough detail here to help you. Are there any details in the Netlify function logs showing a stack trace of what the 500 error is? Can you provide an application to reproduce the issue?

**Reply by akiarostami** [reply]
I just moved this to Vercel. I'll check to see if it works on Netlify after all the updates or not. Will let you know if it didn't. Thank you!

**Reply by akiarostami** [reply]
I finally found out what the problem was It's coming from loading environment variables. It seems that when I load dotenv on the server, I get this error. Perhaps I don't know what I'm doing! 

Is there any good example of how to use environment variables on the server without exposing them to the client?

**Comment by thesyedbasim** [comment]
> I finally found out what the problem was It's coming from loading environment variables. It seems that when I load dotenv on the server, I get this error.
>
> Is there any good example of how to use environment variables on the server without exposing them to the client?

See, when you pulish your app to netlify, or any other hosting platform,
there is no way it can read your private environment variables.

This is because you add .env files to .gitignore so it’s not possible for
hosting providers to know.

Under each project, there will be a setting (in netlify, which is your case
for example) to add environment variables. I am not sure but maybe its
`settings`>`environment variables` in your project dashboard.

So you have to manually add those environmental variables, and redeploy
your site. So this way netlify knows the variables and can use them server
side without exposing to client.

**Reply by akiarostami** [reply]
Thanks @thesyedbasim. I am familiar with how environment variables work. What I'm confused about it VITE's handing of it. Documentation says if I add VITE_ to the beginning of the name, they become available on the client side. And if I don't do that, I can't access the variables anywhere on the server site. It says I need to use dotenv for accessing it on the server, but that's actually what causes the server error. So I'm looking for a simple sample that uses environment variables of both kind, to see the right way of using them in Sveltekit with VITE.

**Reply by benmccann** [reply]
You haven't said what the error is yet. Can you post the logs?
